### DISCUSSION TITLE
Adding an authorization header to the request in server middleware

### DISCUSSION THREAD

**Discussion by Birleyici** [nuxt/nuxt#24014] [Questions]
### Describe the feature

Hello, I spent hours on a process that I hoped would be very simple, but I couldn't find the solution. In a very simple way, I want to add my authorization bearer token to any request in my server middleware file.

```
export default defineEventHandler(async (event) => {

const manager_token = getCookie(event, 'manager_token') || '';

//Here I want to add the access token to the authorization header

...

})
```

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by Birleyici** [comment]
Is this really such a difficult position? The fact that something so simple was so unclear made me question nuxt and its community.

**Comment by naseefpv** [comment]
I use server/api and created a helper '**$apiClient.js**' in **server/utils** folder like this:

```
export default function (event, request, opts) {

    const {
        baseUrl
    } = useRuntimeConfig();

    let token = getCookie(event, "token") || 0

    let headers = {
        "Authorization": "Bearer " + token,
        "Accept": "application/json",
        ...opts?.headers,
    };

    return $fetch(request, {
        baseURL: baseUrl,
        ...opts,
        headers: headers
    })
}
```
then used the above helper instead of **$fetch** in api like this:

```
export default defineEventHandler( async (event) =>{

    const {slug} = event.context.params;

    const query = getQuery(event);

    const data = await $apiClient(event,`product/${slug}`,{
        params:{
            type:query.path
        }
    });

    return data;
})
```

Not sure this is the correct way or not. but it's working fine now. :)
